src/Npgsql/Internal/TypeHandlers/CubeHandler.cs

```diff
@@ -0,0 +1,66 @@
+using System.Linq;
+using Npgsql.BackendMessages;
+using Npgsql.Internal.TypeHandling;
+using NpgsqlTypes;
+
+namespace Npgsql.Internal.TypeHandlers
+{
+    /// <summary>
+    /// A type handler for the PostgreSQL cube data type.
+    /// </summary>
+    /// <remarks>
+    /// See https://www.postgresql.org/docs/current/static/cube.html.
+    ///
+    /// The type handler API allows customizing Npgsql's behavior in powerful ways. However, although it is public, it
+    /// should be considered somewhat unstable, and may change in breaking ways, including in non-major releases.
+    /// Use it at your own risk.
+    /// </remarks>
+    public partial class CubeHandler : NpgsqlSimpleTypeHandler<NpgsqlCube>
+    {
+        #region Constants
+
+        const int DimMask = 0x7fffffff;
+        const uint PointBit = 0x80000000;
+
+        #endregion
+
+        #region Read
+
+        /// <inheritdoc />
+        public override NpgsqlCube Read(NpgsqlReadBuffer buf, int len, FieldDescription? fieldDescription = null)
+        {
+            var header = buf.ReadInt32();
+            var dim = header & DimMask;
```

Comment:
Is there a limit on the number of dimensions? If not, then in theory this is a potentially large data type, and so maybe not appropriate for NpgsqlSimpleTypeHandler.

Reply:
According to the PostgreSQL source code, there is an arbitrary limit of 100 dimensions. I should probably check that when someone creates a cube on the .NET side of things.